apps/managedidentity/managedidentity.go

```diff
@@ -72,8 +73,34 @@ const (
 	imdsEndVar                          = "IMDS_ENDPOINT"
 	msiEndpointEnvVar                   = "MSI_ENDPOINT"
 	identityServerThumbprintEnvVar      = "IDENTITY_SERVER_THUMBPRINT"
+
+	defaultRetryCount = 3
 )
 
+// retry codes for IMDS
+var retryCodesForIMDS = []int{
```

Comment:
This isn't correct. For example, IMDS clients should retry 404. See [IMDS docs](https://learn.microsoft.com/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#error-handling) and [Azure SDK](https://github.com/Azure/azure-sdk-for-go/blob/6bb3a6513654fec9809c4fba235cf9a2c1b7d935/sdk/azidentity/managed_identity_client.go#L108)

Reply:
All the msal's have this `404 or 408 or 429 or 500 or 503 or 504`  
I want to get it right, but the IMDS docs don't have concrete codes only says that `404, 410, 429 and 4xx ` 
If i can get that i will be able to create 2 lists here

Reply:
Let's go with Azure SDK's list @4gust , as that has been tried in production for some time.

Reply:
Azure SDK has a default retry policy for all its services here: https://github.com/Azure/azure-sdk-for-go/blob/6bb3a6513654fec9809c4fba235cf9a2c1b7d935/sdk/azcore/runtime/policy_retry.go#L50 (so retry on 408, 429, 500, 502, 503, 504). 

And for IMDS @chlowell pointed us to it. 

Let's use the same approach - linear retry. I know the IMDS docs suggest exponential backoff, but this is what we discussed in other MSALs and I would like not to diverge here. Let's track any divergence from the other MSALs, otherwise it becomes unscalable. 


CC @neha-bhargava for a perspective on why we chose linear retry in MSALs

Reply:
Linear retry was chosen in MSAL since we do not give flexibility to the user to choose how many times to retry but retry for 3 times (fixed). Azure SDK gives this capability to the user to configure the number of retries. For a small number of retries in MSAL such as 3, it did not make sense to implement exponential backoff.

Azure SDK use their own http client and the retry policy followed in that case is the one that they have configured in their http client so the user experience for them did not change.

Reply:
Also when I checked the Azure SDK .Net code at that time, the exponential backoff computation started after 3 retries with fixed delay. So we decided to keep it linear in MSAL